src/sources/envoy_als/grpc.rs

```diff
@@ -0,0 +1,100 @@
+                                    let mut events = vec![];
+
+                                    if let Some(logs_entries) = msg.log_entries {
+                                        match logs_entries {
+                                            LogEntries::HttpLogs(http_logs) => {
+                                                for l in http_logs.log_entry {
+                                                    let mut evt = LogEvent::default();
+                                                    evt.insert("http_log", Value::from(l));
+
+                                                    if let Some(identifier) = msg.identifier.clone() {
+                                                        stream_identifier = Some(identifier);
+                                                    }
+                                                    if let Some(stream_id) = stream_identifier.clone() {
+                                                        evt.insert("identifier", Value::from(stream_id));
+                                                    }
+
+                                                    events.push(evt);
+                                                }
+                                            },
+                                            LogEntries::TcpLogs(_tcp_logs) => {
+                                                warn!("TCP logs are unsupported at this time.");
+                                                continue;
+                                            }
+                                        }
+                                    }
+
+                                    let count = events.len();
+                                    let byte_size = events.estimated_json_encoded_size_of();
+                                    self.events_received.emit(CountByteSize(count, byte_size));
+
+                                    let resp = self.pipeline
+                                        .clone()
+                                        .send_batch(events)
+                                        .await;
+                                    match resp {
+                                        Ok(()) => {},
+                                        Err(error) => {
+                                            let message = error.to_string();
+                                            emit!(StreamClosedError { error, count });
+                                            return Err(Status::unavailable(message));
+                                        }
+                                    }
+                                },
+                                None => {
+                                    debug!("Sender closed stream");
+                                    break;
+                                }
+                            }
+                        },
+                        Err(err) => {
+                            warn!("error getting stream msg: {}", err);
```

Comment:
Should this be an `Error` internal event? It's unclear from the code if this is an error condition or not. 

```suggestion
                            warn!("error getting message from stream: {}", err);
```


See https://github.com/vectordotdev/vector/blob/master/docs/specs/component.md#componenterror

and https://github.com/vectordotdev/vector/blob/master/docs/specs/instrumentation.md#Error